nodejs/spa/src/main/webapp/index.html

```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Keycloak Single-Page Application Example</title>
+    <script type="text/javascript" src="KC_URL/js/keycloak.js"></script>
+    <script type="text/javascript">
+        function output(content) {
+            if (typeof content === 'object') {
+                content = JSON.stringify(content, null, 2)
+            }
+            document.getElementById('output').textContent = content;
+        }
+
+        function profile() {
+            if (keycloak.idTokenParsed.name) {
+                document.getElementById('name').textContent = 'Hello ' + keycloak.idTokenParsed.name;
+            } else {
+                document.getElementById('name').textContent = 'Hello ' + keycloak.idTokenParsed.preferred_username;
+            }
+            document.getElementById('user').style.display = 'block';
+        }
+
+        keycloak = new Keycloak({ realm: 'quickstart', clientId: 'spa' });
+        keycloak.init({onLoad: 'login-required'}).then(function () {
```

Comment:
Instead of using `.then()` this code can be modernized by using [`await`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await).